specs/interop.md

```diff
@@ -0,0 +1,708 @@
+- The initiating message that corresponds to an executing message MUST come from a chain in its dependency set
+
+Blocks that contain transactions that relay cross domain messages to the destination chain where the
+initiating transaction does not exist MUST be considered invalid and MUST not be allowed by the
+derivation pipeline to be considered safe.
+
+There is no concept of replay protection at the lowest level of abstractions within the protocol because
+there is no replay protection mechanism that fits well for all applications. Users MAY submit an
+arbitrary number of executing messages per initiating message. Applications MUST build their own replay
+protection mechanisms if they are interacting with the lowest level abstractions.
+
+### Depositing an Executing Message
+
+Deposit transactions (force inclusion transactions) give censorship resistance to layer two networks.
+The derivation pipeline must gracefully handle the case in which a user uses a deposit transaction to
+relay a cross chain message. To not couple preconfirmation security to consensus, deposit transactions
+that execute cross chain messages MUST have an initiating message that is considered [safe](#safety) by the remote
+chain's derivation pipeline. This means that the remote chain's data including the initiating message
+MUST be posted to the data availability layer. This relaxes a strict synchrony assumption on the sequencer
+that it MUST have all unsafe blocks of destination chains as fast as possible to ensure that it is building
+correct blocks.
+
+If a deposit transaction references an initiating transaction that is not yet safe or does not exist,
+it MUST be dropped by the derivation pipeline.
+
+This inclusion property prevents a class of attacks where the user can trick the derivation pipeline
+into reorganizing the sequencer.
+
+### Safety
+
+Safety is an abstraction that is useful for reasoning about security. It should be thought about
+as a spectrum from `unsafe` to `final`. Users can choose to operate on information based on its
+level of safety depending on their risk profile and personal preferences.
+
+The following labels are used to describe both inputs and outputs:
+
+- `unsafe`
+- `cross-unsafe`
+- `safe`
+- `finalized`
+
+Inputs correspond to the inputs to the state transition function while outputs correspond to the side
+effects of the state transition function.
+
+Anything before `safe` technically uses a "preconfirmation" based security model which is not part
+of consensus. While useful to have definitions of the default meanings of these terms, they are
+technically policy and may be changed in the future.
+
+The `unsafe` label has the lowest latency while the `finalized` label has the highest latency.
+A set of properties must be matched before an input or an output can be promoted to the next
+label.
+
+The initiating messages for all dependent executing messages MUST be resolved as safe before an L2 block can transition
```

Comment:
typo?
```suggestion
The initiating messages for all destination executing messages MUST be resolved as safe before an L2 block can transition
```

Reply:
Dependent is correct, but it can be worded better. Executing messages depend on specific initiating messages. I'll revisit this after organizing the specs. Merging other typo fixes first now.